api/util/csp-request.ts

```diff
@@ -0,0 +1,85 @@
+import { CostRequest, CspRequestType, CspRequest } from "../../models/cost-jobs";
+import { CspResponse, ProvisionRequest } from "../../models/provisioning-jobs";
+import { getConfiguration } from "../provision/csp-configuration";
+import { getToken } from "../../idp/client";
+import { logger } from "../../utils/logging";
+import axios from "axios";
+
+/**
+ * Make a request to an actual CSP implementation of the ATAT API
+ *
+ * @param request The input request (e.g., provisioning, cost)
+ * @returns the response from the CSP
+ */
+export async function cspRequest(request: CspRequestType<CostRequest | ProvisionRequest>): Promise<CspResponse> {
+  if (!request.body) {
+    return { code: 400, content: { details: "No request body provided" } };
+  }
+  const { targetCsp, portfolioId } = request.body;
+
+  if (!targetCsp || !portfolioId) {
+    logger.error("Request to CSP failed", { input: { targetCsp, portfolioId } });
+    return {
+      code: 400,
+      content: {
+        details: "No Target CSP or the Portfolio Id is not provided.",
+        request: request.body,
+      },
+    };
+  }
+  const baseUrl = (await getConfiguration(targetCsp.name))?.uri;
+  let url: string;
+
+  switch (request.requestType) {
+    case CspRequest.PROVISION:
+      url = `${baseUrl}/portfolios`;
+      break;
+    case CspRequest.COST:
+      url = `${baseUrl}/portfolios/${portfolioId}/cost`;
+      break;
+  }
+
+  if (!baseUrl || !baseUrl.startsWith("https://")) {
+    logger.error("Invalid CSP configuration (Demo)", {
+      input: {
+        csp: targetCsp.name,
+      },
+      resolvedUrl: url,
+      configSecretPath: process.env.CSP_CONFIG_SECRET_NAME,
+    });
+    return {
+      code: 400,
+      content: {
+        details: "Invalid CSP provided",
+        request: request.body,
+      },
+    };
+  }
+
+  const response = await axios.post(url, request.body, {
+    headers: {
+      Authorization: `Bearer ${(await getToken()).access_token}`,
+      "User-Agent": "ATAT v0.2.0 client",
+    },
+    // Don't throw an error on non-2xx/3xx status code (let us handle it)
+    validateStatus() {
+      return true;
+    },
+  });
+
+  if (response.status !== 200 && response.status !== 202) {
```

Comment:
```suggestion
  if (![200, 202].includes(response.status)) {
```

Thoughts?

Reply:
I like it, we can add/remove as needed without much change. If it gets bigger we can move into a variable, but I like it this way.